torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -49,6 +49,11 @@ void copy_range(variable_list& out, IndexRange range, at::ArrayRef<Tensor> t) {
   std::copy(t.begin(), t.end(), out.begin() + range.first);
 }
 
+Tensor copysign_tensor_backward(Tensor grad, Tensor self, Tensor other) {
+  auto result = grad * self.sign() * other.sign();
```

Comment:
I am not sure that you need the `self.sign()` here.
It will fail gradcheck when you add it to the list in common_method_invocation.py if the formula is wrong. But I think you will need to change this.

Reply:
For instance:
- `a = tensor(-1.)`
- `b = tensor(1.)`
- `c =  torch.copysign(a, b) = tensor(1)`

The derivative of `a` is `-1` rather than `b.sign() = 1`. Any thought on that?

Reply:
Ho right, the derivative is -1 when we change and 1 otherwise so you need both! Agree with you.

Also there is a corner case at `0` here where `sign()` returns `0`. What is `copysign()` doing for that? Is the backward formula good for this case as well?

Reply:
I have actually tested the following cases:
- `copysign(0, 1) = 0` -> grad = 0
- `copysign(0, -1) = 0` -> grad = 0
- **`copysign(1,** 0) = 1` -> grad = 1**
- **`copysign(-1, 0) = 1` -> grad = -1**

The last two cases are not correct for this implementation.